### PR TITLE
Add tests to operator/pkg/httprequest and helm

### PR DESCRIPTION
**Pull request by adiprerepa**

### CODE REVIEW

**Review by dgn — COMMENTED** [review]
thank you! I left some comments

**Review by dgn — COMMENTED** [review]
thank you! I left some comments

File: operator/pkg/helm/fs_renderer_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			if fileTemplateRenderer := NewFileTemplateRenderer(tt.inHelmChartDirPath, tt.inComponentName, tt.inNamespace); *fileTemplateRenderer != tt.want {
+				t.Errorf("%s, want: %+v, got: %+v", tt.desc, tt.want, *fileTemplateRenderer)
+			}
+		})
+	}
+}
+
+// TODO(adiprerepa): find a way to pass a chart in and test it.
```
[the remark this answers is not in the document]
Comment (adiprerepa) [reply operator/pkg/helm/fs_renderer_test.go:68]: I see. The way I understand it, the function takes in yaml values and renders the manifest. Is this manifest like a standard yaml file? And what exactly is a chart? Sorry, I'm new to all this.

File: operator/pkg/helm/fs_renderer_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 Istio Authors
```
[the remark this answers is not in the document]
Comment (adiprerepa) [reply operator/pkg/helm/fs_renderer_test.go:1]: done

File: operator/pkg/httprequest/httprequest_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Istio Authors
```
[the remark this answers is not in the document]
Comment (adiprerepa) [reply operator/pkg/httprequest/httprequest_test.go:1]: done

File: operator/pkg/httprequest/httprequest_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+func TestGet(t *testing.T) {
+	expected := "fooey-baroque"
+	url := "/fooey/baroque/cazoo"
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		if req.URL.String() != url {
+			t.Errorf("Made request to wrong URL, want: %s, got: %s", url, req.URL.String())
+		}
+		rw.Write([]byte("fooey-baroque"))
```
[the remark this answers is not in the document]
Comment (adiprerepa) [reply operator/pkg/httprequest/httprequest_test.go:30]: good catch - done

**Comment by adiprerepa** [comment]
I'm Still having trouble figuring it out. If you think the rest is ok, if you merge it I would be able to take another attempt at it. thanks. @dgn

**Comment by adiprerepa** [comment]
@dgn I figured out how to load the file. Is the rest OK?

File: operator/pkg/httprequest/httprequest_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	}))
+	defer server.Close()
+	response, err := Get(server.URL + url)
+	if err != nil {
+		t.Errorf("Unexpected Error In Making Request: %s", err.Error())
+	}
+	if expected != string(response) {
+		t.Errorf("Returned unexpected response, want: %s, got: %s", expected, string(response))
+	}
+}
```
Comment (morvencao) [inline operator/pkg/httprequest/httprequest_test.go:40]: Can you make this to table driven tests so that we can easily add more test cases?
Others
/lgtm

Comment (adiprerepa) [reply operator/pkg/httprequest/httprequest_test.go]: @morvencao i refactored the tests.

**Review by morvencao — APPROVED** [review]

**Pull request merged**